mood-analysis.rb

```diff
@@ -28,5 +35,49 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+
+  text.each do |entry|
+  puts "#{entry[0..4]} #{analyze_mood(entry)}"
+  end
+
+  #determine how many logged entries it takes until there have been three :-) happy days.
+  def happy_days(text)
+    good_days = 0
+    day_count = 0
+    text.each do |text|
+      if good_days < 3 #if there aren't at least 3 happy days?
+        day_count += 1
+        if analyze_mood(text) == ":-)"
+          good_days += 1
+        end
+      end
+    end
+    puts "It takes #{day_count} days for 3 happy days to occur."
+  end
+
+  #to determine the most common mood across all logged entries
+  def overall_mood(text)
+    happy = 0
+    sad = 0
+    meh = 0
+      text.each do |entry|
+        if analyze_mood(entry) == ":-)"
+          happy += 1
+        elsif analyze_mood(entry) == ":-("
+          sad += 1
+        else
+          meh += 1
+        end
+      end
+    if happy > sad
+      puts "The most common mood is :-)"
+    elsif happy < sad
+      puts "The most common mood is :-("
+    else sad == happy || meh == 0
+      puts "The most common mood is :-|"
+    end
+  end
```

Comment:
This method is okay for the most part, but not really scalable and doesn't handle :-| quite right. Using a hash here would have been the ideal solution because we could track the emoticon and it's associate count within the text as well. This is actually a quite common type of operation for algorithms or whiteboarding problems, so one you should understand even if it wasn't your first thought.

Here's my code:
```
def overall_mood(entries)
  return nil if entries.length == 0
  emoticons = Hash.new(0)
  entries.each do |entry|
    emoticon = analyze_mood(entry)
    emoticons[emoticon] += 1
  end
  return emoticons.max_by{|k,v| v}[0]
end
```

Take a look at that and see if it makes sense. If it doesn't, I'll hopefully be going over this in class, or we can chat 1-1.